### PR TITLE
Fslc adin1110 backport

### PR DESCRIPTION
**Pull request by tachicialex**
4e2b2cda4889 - arch: arm64: boot: dts: freescale: Add DT for ADIN1110
052a625daeac - dt-bindings: net: adin1110: Add docs
4fd814896524 - net: ethernet: adi: adin1110: Fix notifiers
fb457b7cc41b - net: ethernet: adi: adin1110: Add check in netdev_event
66a292befd03 - net: ethernet: adin1110: Add missing MODULE_DEVICE_TABLE
ad962e0f17a5 - net: ethernet: adi: Fix return value check in adin1110_probe_netdevs()
029001fdacf2 - net: ethernet: adi: Fix invalid parent name length
e3d10f7b1dbe - net: ethernet: adi: Add ADIN1110 support

ADIN1110/2111 driver has been accepted upstream and with this PR I want to add support to ADIN1110/2111 in your version of 5.19 also. Here at ADI we had some requests for a branch (Driver + DT) to use this product with the 8MMINI-BB SoC board.

### CODE REVIEW

**Comment by zandrey** [comment]
Hello @tachicialex 

Unfortunately, I do not see those patches in the Linus tree yet. 

Only patches that have reference to `adin1100` that could be seen integrated are those in the following list (excluding merge commit):
- 875b718ac380 `("net: phy: adin1100: add PHY IDs of adin1110/adin2111")`
- 48f20f902119 `("net: phy: adin1100: Add SQI support")`
- 7eaf9132996a `("net: phy: adin1100: Add initial support for ADIN1100 industrial PHY")`

Last 2 patches are already present in the `linux-5.19.y` and `5.19.x+fslc` branches. Where was this series applied?

Moreover, from commit lore links I do not see that anyone had a look at those patches yet, so I cannot say 100% they are accepted in upstream.

Can you please clarify the state of those patches?

In general, the branch you're targeting shall have all patches submitted in upstream, with only exceptions of crucial bugfixes to be applied directly. It looks like you're trying to submit a rather feature here, which shall land in upstream first.

-- andrey

**Comment by tachicialex** [comment]
Hi @zandrey ,
Thanks for taking the time.
They are on the Linux Kernel  [net-next](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/log/?qt=grep&q=adin1110) and [rc version]( https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/log/?h=v6.1-rc5&qt=grep&q=adin1110). 

The adin1100 is the PHY driver. The adin1110.c referes to the adin1110 and adin2111 MAC-PHY
and switch that make use of the adin1100 phy driver (internal mdio bus).

They have undergone review by the maintainers from net and dt list.
If you want we could wait for them to be merged in an actual version release (no rc stuff)
and if they don't get any comments merge them here?

-Alexandru

**Comment by zandrey** [comment]
@tachicialex 

Thanks for the explanation! I have to admit, I did overlook the part number and found only PHY patches.

Still the list is not complete, as Linus tree has only those in (MAC+PHY):
21ce2c121fa0 `("net: ethernet: adi: adin1110: Fix notifiers")`
f93719351b0e `("net: ethernet: adi: adin1110: Add check in netdev_event")`
bb65131bb62c `("net: ethernet: adin1110: Add missing MODULE_DEVICE_TABLE")`
9f1e337851be `("net: ethernet: adi: Fix return value check in adin1110_probe_netdevs()")`
9fd12e869e17 `("dt-bindings: net: adin1110: Add docs")`
bc93e19d088b `("net: ethernet: adi: Add ADIN1110 support")`
875b718ac380 `("net: phy: adin1100: add PHY IDs of adin1110/adin2111")`
48f20f902119 `("net: phy: adin1100: Add SQI support")`
7eaf9132996a `("net: phy: adin1100: Add initial support for ADIN1100 industrial PHY")`

All of those would be picked up into new `6.1.x+fslc` once _Linux 6.1_ is out.

Which leaves those 2 patches not (yet) applied upstream:
4e2b2cda4889 - arch: arm64: boot: dts: freescale: Add DT for ADIN1110
029001fdacf2 - net: ethernet: adi: Fix invalid parent name length

IMO, new DTB shall go upstream, as it would require Maintainer entry. But I would leave it up to @otavio if it can be applied directly.

I would also suggest that you re-base the PR onto `6.0.x+fslc` branch as `5.19.x+fslc` is already EOL.

-- andrey

**Comment by otavio** [comment]
Yes, please rebase this on top of new branch so we avoid putting new stuff in old kernels.

**Comment by tachicialex** [comment]
Rebased everything on the 6.0.x+fslc branch.

**Comment by otavio** [comment]
@tachicialex have you sent the missing patches for LKML?

**Comment by tachicialex** [comment]
> @tachicialex have you sent the missing patches for LKML?
This one is sent already:
https://github.com/Freescale/linux-fslc/commit/029001fdacf203a46a0e5dcdef2c1b211ee76e7c - net: ethernet: adi: Fix invalid parent name length

Will be dropping the custom dts. It is way too specific. Will post it on our how to setup wiki page. Thanks a lot again for the time.

**Review by zandrey — APPROVED** [review]

**Pull request merged**
